### ISSUE TITLE
PoE-Antiquary.xyz integration

### ISSUE THREAD

**Issue by CBrach** [PoE-TradeMacro/POE-TradeMacro#702]
Hi,
I'm the dev of poe-antiquary.xyz . I created a small custom macro to open the item page on antiquary when hovering over an item and pressing ctrl+alt+a. You can find the code in the attachments.

CM_ExecuteCustomMacrosCode_Label:   
    Hotkey, ^!a, CM_PoEAntiquary_Label
Return

[PoEAntiquaryMacro.txt](https://github.com/PoE-TradeMacro/POE-TradeMacro/files/1887514/PoEAntiquary.txt)

I don't know a lot about AHK, so I'd be very thankful if a contributor could take a quick look over the code if I made any big errors (code is working fine afaik). Also, I couldn't figure out how to create the macro as a build-in feature to submit a pull request - would be very grateful if one of you could briefly explain where to put the code or add it himself.

Thanks and have a nice week
Chris

**Comment by Eruyome** [comment]
Sorry for the late answer, took some time off.

I'll take a look at it and I could add it as part of addtional macros, there's a hotkey that I want to replace anyway, there's not really any point in supporting http://www.pathof.info/ anymore.

**Comment by CBrach** [comment]
The GUI triggers for items that have more than one variation (e.g. Vinktar, the Abyss armour items with 1/2 sockets, ...). 
The macro uses an API endpoint to get the internal id from the copied item name. This endpoint returns an array with all items matching the name. If more than one item is in this array, the GUI is opened and asks the user to chose a variant.
```

global AntiquaryData := JSON.Load(JSONFile)
length := AntiquaryData.Length()
if(length == 1){
    data := AntiquaryData[1]
    id := data["id"]
    name := data["name"]
    AntiquaryOpenBrowser(AntiquaryType, name, id)
}
if(length > 1){
    AntiquaryCreateGUI()
}
```

**Comment by Eruyome** [comment]
Yeah, noticed that.

Can you add an endpoint to return the last league that your're supporting? I don't like this hardcoding here:

```
	league := TradeGlobals.Get("LeagueName")
	If (RegExMatch(league, "Hardcore.*")) {
		league := "Abyss HC"
	} Else{
		league := "Abyss"
	}
```

**Comment by Eruyome** [comment]
One more thing, there isn't any page that lists all variants of an item, like Impresence/Vessel of Vinktar, right?
I, as a user, would prefer being directed to such a page instead of having to use an AHK gui to select the variant. I believe it's faster to go there and click the right link than having to use an AHK dropdown menu, and more convenient. It may be nice to have for your site also.

I feel the Gui is an unnecessary intermediate step and it popping up an unexpected behaviour since most of the time you won't see it.

**Comment by CBrach** [comment]
Hi,
sorry for the late answer, been a busy week.

Thank you for your suggestions, I definitely agree with all of them. I've already implemented the necessary changes them locally and will upload them later today. 

The relevant API changes:
```
poe-antiquary.xyz/api/macro/{item_type_name}/{item_name}

Returns:

JSON encoded Array
[
    league: {latest league_name}
    name: {passed item_name}
    itemType: {passed itemType_name}
    items: [
        [
            id: {item_id}
            name: {item_name}
        ]
    ]
]
```

There are three possible outcomes of an API request that need to be handled:

- the items array is empty 
`No item found, display error message`
- the items array contains exactly 1 item 
`Open browser (poe-antiquary.xyz/{league_name}/{item_type_name}/{item_name}/{item_id})`
- the items array contains >1 item
`Open browser (poe-antiquary.xyz/{league_name}/{item_type_name}?name={item_name}`

I will upload a new AHK script later once the changes have been pushed to the remote server.

**Comment by Eruyome** [comment]
As long as I know all the API calls you don't need to make and upload a new AHK script once the changes are live.

**Comment by CBrach** [comment]
I just pushed the changes to the server. I made no changes to the API mentioned earlier, so the endpoint for the macro is still
`poe-antiquary.xyz/api/macro/{item_type_name}/{item_name}`

Here's an exemplary API request + response:

`http://dev.poe-antiquary.xyz/api/macro/Armour/Bubonic%20Trail`

```
{
    "league":"Abyss",
    "name":"Bubonic Trail",
    "itemType":"Armour",
    "items":
        [
            {
                "id":4771,
                "name":"Bubonic Trail, 2 Jewels"
            },
            {
                "id":4766,
                "name":"Bubonic Trail, 1 Jewel"
            }
        ]
}
```
In this case, the macro should open `poe-antiquary.xyz/Abyss/Armour?name=Bubonic Trail` as the Response contains two items. 
Currently enabled item types are:
- Currency
- Fragment
- Essence
- Divination
- Prophecy
- Jewel
- Flask
- Weapon
- Armour
- Accessory
- Map
- Unique Map

Please let me know if you prefer a different API syntax.

**Comment by Eruyome** [comment]
Everythings seems to work but there's a small issue with maps. For example using it on "Academy Map" returns
```
           [items] => Array
                (
                    [1] => Array
                        (
                            [id] => 19
                            [name] => Academy Map (<3.1)
                        )
                    [2] => Array
                        (
                            [id] => 2395
                            [name] => Academy Map
                        )
                )
```
And then opens a page where you can see a normal academy map and the shaped version. I believe that's a bit unexpected. The first item in the returned array is probably the old academy map, which is not displayed at all on the page, while the shaped version is not included but displayed.

http://poe-antiquary.xyz/Abyss/Map?name=Academy%20Map

**Comment by CBrach** [comment]
I fixed #1, the API no longer returns <3.1 items. 
Displaying the shaped version when searching for Academy in the item table is working as intended as it's using the same data table filter as the filter in the DOM.
